test/unit/allrules.spec.js

```diff
@@ -0,0 +1,119 @@
+
+/*
+ * GNU AGPL-3.0 License
+ *
+ * Copyright (c) 2021 - present core.ai . All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along
+ * with this program. If not, see https://opensource.org/licenses/AGPL-3.0.
+ *
+ */
+
+// jshint ignore: start
+/*global describe, it*/
+
+import * as chai from 'chai';
+let expect=chai.expect;
+import {default as allRules} from '../../src/rules/allRules.js';
+import 'mocha';
+
+const subkeys=new Set();
+for(var i in allRules){
+    var key = i;
+    var val = allRules[i];
+    for(var j in val){
+        var sub_key = j;
+        var sub_val = val[j];
```

Comment:
Use proper names for variables instead of I &J

Reply:
Ok i will change it.